xtc-history/xtc-history-common/src/bucket.rs

```diff
@@ -132,7 +132,7 @@ impl<Address, Event> BucketData<Address, Event> {
 
         let take = limit + 1;
         let end = (offset - bucket_offset) as usize;
-        let start = end.checked_sub(take).unwrap_or(0);
+        let start = end.saturating_sub(take);
```

Comment:
just as a sidenote: we currently set a compiler flag to panic on integer overflow/underflows, as the code almost never checks for those manually

Reply:
So should we leave it as is or remove it?

Reply:
We can keep saturating_sub. Ideally we would have never had to switch on release mode int checking, but the Dank code almost never checks for overflow/underflow manually.